pandas/_testing.py

```diff
@@ -619,7 +619,10 @@ def _get_ilevel_values(index, level):
         # accept level number only
         unique = index.levels[level]
         level_codes = index.codes[level]
-        filled = take_1d(unique.values, level_codes, fill_value=unique._na_value)
+        if is_extension_array_dtype(unique):
```

Comment:
this is not as nice (though its only for testing), won't unique.take(...) work here?

Reply:
ill give it a shot

Reply:
Turns out this unique.take(...) wont work here.  the trouble is that we are calling _shallow_copy below (presumably for perf) and using unique.take ends up passing the wrong types to shallow_copy

Reply:
> using unique.take ends up passing the wrong types to shallow_copy

then i'd rather not change this, take_1d just does the right thing, so why change it?

Reply:
Looks like passing unique._values instead of unique.values fixes the problem cleanly

Reply:
ok great